lib/integrations/awsoidc/aws_app_access_iam_config.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package awsoidc
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/gravitational/trace"
+
+	awslib "github.com/gravitational/teleport/lib/cloud/aws"
+)
+
+const (
+	// defaultPolicyNameForAWSAppAccess is the default name for the Inline Policy added to the IntegrationRole.
+	defaultPolicyNameForAWSAppAccess = "AWSAppAccess"
+)
+
+// AWSAppAccessConfigureRequest is a request to configure the required Policies to use AWS App Access.
+type AWSAppAccessConfigureRequest struct {
+	// Region is the AWS Region.
+	// Used to set up the AWS SDK Client.
+	//Region string
+
+	// IntegrationRole is the Integration's AWS Role used to set up Teleport as an OIDC IdP.
+	IntegrationRole string
+
+	// IntegrationRoleAWSAppAccessPolicy is the Policy Name that is created to allow access to call AWS APIs.
+	// Defaults to AWSAppAccess
+	IntegrationRoleAWSAppAccessPolicy string
+}
+
+// CheckAndSetDefaults ensures the required fields are present.
+func (r *AWSAppAccessConfigureRequest) CheckAndSetDefaults() error {
+	if r.IntegrationRole == "" {
+		return trace.BadParameter("integration role is required")
+	}
+
+	if r.IntegrationRoleAWSAppAccessPolicy == "" {
+		r.IntegrationRoleAWSAppAccessPolicy = defaultPolicyNameForAWSAppAccess
+	}
+
+	return nil
+}
+
+// AWSAppAccessConfigureClient describes the required methods to create the IAM Policies required for AWS App Access.
+type AWSAppAccessConfigureClient interface {
+	// PutRolePolicy creates or replaces a Policy by its name in a IAM Role.
+	PutRolePolicy(ctx context.Context, params *iam.PutRolePolicyInput, optFns ...func(*iam.Options)) (*iam.PutRolePolicyOutput, error)
+}
+
+type defaultAWSAppAccessConfigureClient struct {
+	*iam.Client
+}
+
+// NewAWSAppAccessConfigureClient creates a new AWSAppAccessConfigureClient.
+func NewAWSAppAccessConfigureClient(ctx context.Context) (AWSAppAccessConfigureClient, error) {
```

Comment:
Ditch the interface and return a concrete type.

A good rule of thumb is take interfaces as parameters (if necessary!) and return concrete types.

- https://go.dev/wiki/CodeReviewComments#interfaces